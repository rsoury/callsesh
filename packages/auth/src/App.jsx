import React, { useEffect, useState } from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import { matchPath, withRouter } from "react-router";
import RouterPropTypes from "react-router-prop-types";
import { useStyletron } from "baseui";
import { ToasterContainer, PLACEMENT } from "baseui/toast";

import Header from "@/components/Header";
import { shouldSignup } from "@/utils/auth";

import Signup from "./Signup";
import Login from "./Login";

const App = ({ location }) => {
	const [redirect, setRedirect] = useState("");
	const [css] = useStyletron();

	useEffect(() => {
		// Redirect to signup if action signup and current route is not signup.
		if (shouldSignup()) {
			const match = matchPath(location.pathname, {
				path: "/signup",
				exact: true
			});
			if (!match) {
				setRedirect("/signup");
			}
		}
	}, []);

	if (redirect) {
		return <Redirect to={redirect} />;
	}

	return (
		<ToasterContainer
			placement={PLACEMENT.topRight}
			autoHideDuration={10000}
			overrides={{
				Root: {
					style: {
						zIndex: "9999999"
					}
				}
			}}
		>
			<main
				id="wagecall-auth"
				className={css({
					display: "flex",
					justifyContent: "center",
					alignItems: "center",
					minHeight: "100%",
					width: "100%",
					maxWidth: `640px`,
					margin: "0 auto",
					position: "relative"
				})}
			>
				<div className={css({ width: "100%" })}>
					<Header />
					<Switch>
						<Route exact path="/signup">
							<Signup />
						</Route>
						<Route exact path="/">
							<Login />
						</Route>
						<Route path="*">
							<Redirect to="/" />
						</Route>
					</Switch>
				</div>
			</main>
		</ToasterContainer>
	);
};

App.propTypes = {
	location: RouterPropTypes.location.isRequired
};

export default withRouter(App);
