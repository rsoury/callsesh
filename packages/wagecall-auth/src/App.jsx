import React from "react";

const App = () => {
	return (
		<div>
			<div>Hello World</div>
		</div>
	);
};

export default App;
