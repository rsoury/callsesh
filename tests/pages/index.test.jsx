import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import App from "@/pages/index";

describe("With Enzyme", () => {
	it("App shows content", () => {
		const app = shallow(<App />);

		expect(app.find("div").text().length).toBeGreaterThan(0);
	});
});

describe("With Snapshot Testing", () => {
	it("App shows matches snapshot", () => {
		const component = renderer.create(<App />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
