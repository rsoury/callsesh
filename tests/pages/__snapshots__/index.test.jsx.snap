// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello, Sunshine!" 1`] = `
<div>
  <h1
    className=""
  >
    Title
  </h1>
  <h2
    className=""
  >
    Subtitle
  </h2>
  <p
    className=""
  >
    Styled by hook
  </p>
</div>
`;
